### ISSUE TITLE
[bug] passwords with special characters not properly encoded when checking out sources from SCM

### ISSUE THREAD

**Issue by DukeOfKirkcaldy** [conan-io/conan#8224]
Authentication started failing after upgrading to 1.32.0

### Environment Details (include every applicable attribute)
  * Operating System+version:
  * Compiler+version:
  * Conan version: 1.32.0
  * Python version:

### Logs (Executed commands with output) (Include/Attach if Applicable)
command:
```
conan install my_package/1.0.0@user/channel#f6c6506327598117fef706344bbf6bb11bcb8f3c --build=my_package/1.0.0@user/channel#f6c6506327598117fef706344bbf6bb11bcb8f3c --build=missing --lockfile=conan.lock --lockfile-out=updated_conan.lock 
```
output:
```
my_package/1.0.0@user/channel: SCM: Getting sources from url: 'https://my/repo/'
ERROR: Couldn't checkout SCM: Command 'git -c http.sslVerify=true clone "https://****:****@my/repo/" .   ' returned non-zero exit status 128.
Cloning into '.'...
remote: Unauthorized
fatal: Authentication failed for 'https://my/repo/'
```

**Comment by memsharded** [comment]
This could come from https://github.com/conan-io/conan/pull/8016

**Comment by memsharded** [comment]
I am trying to add a unittest with special characters, like:

```python
    def test_url_with_user_password_characters_param(self):
        scm = SCMBase(username="user", password="p.-+*¨Ç{}%&$()9")
        self.assertEqual('https://user:p.-+*¨Ç{}%&$()9@github.com/conan-io/conan.git',
                         scm.get_url_with_credentials("https://github.com/conan-io/conan.git"))
```
But this works fine. Can you please detail how are you declaring your scm, and how are you passing the password?

**Comment by DukeOfKirkcaldy** [comment]
Hi @memsharded,

I'm passing the username and password through environmental variables like so:

_conanfile.py_

```
    scm = {
        "type": "git",
        "url": "auto",
        "revision": "auto",
        "username": tools.get_env("GIT_USER"),
        "password": tools.get_env("GIT_PASS")
    }
```

_Jenkinsfile_
```
withCredentials ([usernamePassword(credentialsId: 'jenkins_git_creds', \
                                   passwordVariable: 'GIT_PASS', \
                                   usernameVariable: 'GIT_USER')]) {
    command (
        label: "Build ${ref}",
        script: "conan install ${ref} --build=${ref} --build=missing --lockfile=${lockfile} --lockfile-out=${updated_lockfile}"
    )
 }
```

**Comment by DukeOfKirkcaldy** [comment]
Just a quick update.  

Before 1.32.0, the password was provided url-encoded (e.g. `%` to `%25`) and with OS special characters escaped. (e.g. `%` to `%%`)

This breaks in 1.32.0.  It seems removing the OS escapes solves the issue.

Can you please advise how passwords should be provided?

I would still consider this to be a breaking change since CI scripts must change their behavior depending on installed conan version on the build machine.

**Comment by memsharded** [comment]
Closed in #8355
